cardinal/main.go

```diff
@@ -29,47 +29,42 @@ func main() {
 	// Register components
 	// NOTE: You must register your components here,
 	// otherwise it will show an error when you try to use them in a system.
-	err := errors.Join(
-		cardinal.RegisterComponent[component.PlayerComponent](world),
-		cardinal.RegisterComponent[component.HealthComponent](world))
-	if err != nil {
-		log.Fatal().Err(err)
-	}
+	Must(
+		cardinal.RegisterComponent[component.Player](world),
+		cardinal.RegisterComponent[component.Health](world),
+	)
 
 	// Register transactions
 	// NOTE: You must register your transactions here,
 	// otherwise it will show an error when you try to use them in a system.
-	err = cardinal.RegisterTransactions(world,
+	Must(cardinal.RegisterTransactions(world,
 		tx.CreatePlayer,
 		tx.AttackPlayer,
-	)
-	if err != nil {
-		log.Fatal().Err(err)
-	}
+	))
 
 	// Register read endpoints
 	// NOTE: You must register your read endpoints here,
 	// otherwise it will not be accessible.
-	err = cardinal.RegisterQueries(world,
+	Must(cardinal.RegisterQueries(world,
 		query.Constant,
-	)
-	if err != nil {
-		log.Fatal().Err(err)
-	}
+	))
 
 	// Each system executes deterministically in the order they are added.
 	// This is a neat feature that can be strategically used for systems that depends on the order of execution.
 	// For example, you may want to run the attack system before the regen system
 	// so that the player's HP is subtracted (and player killed if it reaches 0) before HP is regenerated.
-
 	cardinal.RegisterSystems(world,
 		system.AttackSystem,
 		system.RegenSystem,
-		system.PlayerSpawnerSystem)
+		system.PlayerSpawnerSystem,
+	)
 
-	err = world.StartGame()
+	Must(world.StartGame())
```

Comment:
the "world" here is a *cardinal.World, and the "world" below is a cardinal.WorldContext. I think duplicating the name could cause confusion. I'd suggest using `worldCtx` below.

Reply:
renamed this to `w` instead. i really would prefer to preserve `world` in system because that is the least ugly